packages/framework/src/Model/Article/ArticleFacade.php

```diff
@@ -143,12 +143,21 @@ public function edit($articleId, ArticleData $articleData)
         $article = $this->articleRepository->getById($articleId);
         $originalName = $article->getName();
 
+        // the intent is to save nothing in seo_meta_description column
         $articleData->seoMetaDescription = null;
 
-        if (count($articleData->urls->toDelete[1]) > 0) {
+        // error 500 thrown when trying to save changes after a certain url was marked for deletion
+        if (isset($articleData->urls->toDelete[1]) && count($articleData->urls->toDelete[1]) > 0) {
             throw new Exception();
         }
 
+        // disable text alignment in the wysiwyg editor
+        $articleData->text = preg_replace(
```

Comment:
I think you should also put the same logic into the `create` method - otherwise, the intended bug will not be there when a new article is created. (And please, DRY, a private method for this would be sexy :grin:) 

The same applies to the meta description modification (you are making it null only here, not in the edit method). It is true that this is solved in `Article::getSeoMetaDescription` method, but it is still inconsistent:
- when creating a new article, SEO meta description is saved into the database, but `null` is returned from the getter
- when editing an article, null is saved into the database
So maybe we can just keep the getter and do not modify the input if we do not mind the value in the database, or we should modify the value in both `create` and `edit` methods :slightly_smiling_face: 

Long story short, it would be nice to unify the approach - if we do not mind the original data in the database, I think the easiest way is just modifying the getters. We can discuss it further if you want, let me know :wink: